### PR TITLE
Add keymaps for myself

### PR DESCRIPTION
**Pull request by toshi0383**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace addition

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/reviung39/keymaps/toshi0383/readme.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# The default keymap for REVIUNG39
```
Comment (noroadsleft) [inline keyboards/reviung39/keymaps/toshi0383/readme.md]:
```suggestion
# toshi0383's keymap for REVIUNG39
```

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+// entirely and just use numbers.
+enum layer_number {
+    _QWERTY = 0,
+    _LOWER,
+    _RAISE,
+    _ADJUST,
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
```
Comment (noroadsleft) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: You don't need this custom keycode, as you only have the QWERTY base layer (no Colemak/Dvorak/etc.).

```suggestion
  LOWER = SAFE_RANGE,
```

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+  LOWER,
```
Comment (noroadsleft) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]:
```suggestion
```

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+};
+
+enum macro_keycodes {
+  KC_SAMPLEMACRO,
+};
+
+
+//Macros
+#define M_SAMPLE M(KC_SAMPLEMACRO)
+
```
Comment (noroadsleft) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: You don't need Lines 25-32 at all. They should be removed.

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        set_single_persistent_default_layer(_QWERTY);
+      }
+      return false;
+      break;
+
```
Comment (noroadsleft) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: You don't need Lines 113-119. They should be deleted.

File: keyboards/zinc/keymaps/toshi0383/rules.mk [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+CONSOLE_ENABLE = no         # Console for debug(+400)
+COMMAND_ENABLE = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no      # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
+SWAP_HANDS_ENABLE = no        # Enable one-hand typing
```
Comment (noroadsleft) [inline keyboards/zinc/keymaps/toshi0383/rules.mk]: Could you please change Lines 1-18 to:

```Makefile
# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Virtual DIP switch configuration
MOUSEKEY_ENABLE = no        # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
MIDI_ENABLE = no            # MIDI controls
AUDIO_ENABLE = no           # Audio output on port C6
UNICODE_ENABLE = no         # Unicode
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.  Do not enable this with audio at the same time.
SWAP_HANDS_ENABLE = no      # Enable one-hand typing
```

This removes the notes about firmware size impacts (they're basically never accurate) and aligns the comments.

Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/rules.mk]: done in 1f3858c2f

File: keyboards/zinc/keymaps/toshi0383/keymap.c (hunk 5 above)
Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: 0299bb0c39851c4b834b4a3ad31fde9cb207bed6

File: keyboards/zinc/keymaps/toshi0383/keymap.c (hunk 4 above)
Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: 0299bb0c39851c4b834b4a3ad31fde9cb207bed6

**Comment by toshi0383** [comment]
@noroadsleft Thanks for the review☺️

File: keyboards/reviung39/keymaps/toshi0383/keymap.c [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+  
+  [_ADJUST] = LAYOUT_reviung39(
+    RGB_VAI,   RGB_SAI, RGB_HUI,  RGB_MOD,  XXXXXXX,   RGB_TOG,            KC_Y,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    RGB_VAD,   RGB_SAD, RGB_HUD,  RGB_RMOD, XXXXXXX,   XXXXXXX,            XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+    XXXXXXX,   XXXXXXX, XXXXXXX,  XXXXXXX,  XXXXXXX,   RESET,            XXXXXXX,    XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,  XXXXXXX,
+                                                       _______,  XXXXXXX,  _______
+  ),
+};
+
+uint32_t layer_state_set_user(uint32_t state) {
```
Comment (drashna) [inline keyboards/reviung39/keymaps/toshi0383/keymap.c]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
```

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 8]
```diff
@@ -0,0 +1,147 @@
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: This line is not actually needed, and should be removed. 

```suggestion
```

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+      _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______,  _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, RESET,  _______, _______, _______, _______, _______, _______ \
+      ),
+
+};
+
+// define variables for reactive RGB
+bool TOG_STATUS = false;
```
Comment (drashna) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: Wait, what are you trying to do there?

Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: Oh thank you that line is unnecessary too.
I copied the default keymap and removed LED related code.

Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: I’ll remove this line!

And I also have issue not able to enter ADJUST layer by RAISE + N.
Could someone please help debugging?

Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: Oh the issue is on reviung39 keymap, not zinc.

File: keyboards/reviung39/keymaps/toshi0383/keymap.c [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+   KC_TILD, KC_EXLM, KC_AT, KC_HASH, KC_DLR,  KC_PERC,    KC_CIRC, KC_AMPR, KC_ASTR, _______, _______, _______,
+   _______, KC_LT, KC_GT, KC_LPRN, KC_RPRN, KC_GRV,       KC_MINS, KC_LCBR, KC_RCBR, KC_LBRC, KC_RBRC, KC_PIPE,
+   _______, _______, _______, _______, KC_EQL, KC_QUES,   KC_UNDS, KC_PLUS, KC_COMM, KC_DOT, KC_SLSH, _______,
+                                      _______,  KC_ENT,   KC_RGUI
+  ),
+  
+  [_RAISE] = LAYOUT_reviung39(
+  _______, KC_1,    KC_2,    KC_3,    KC_4,    KC_5,      KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  _______,
+  _______, KC_COLN, KC_SCLN, KC_DQT, KC_QUOT, KC_BSLS,    KC_LEFT, KC_DOWN, KC_UP, KC_RGHT, _______, _______,
+  _______, _______, _______, _______, _______, _______,   ADJUST, _______, _______, _______, _______, RGB_MOD,
```
Comment (toshi0383) [inline keyboards/reviung39/keymaps/toshi0383/keymap.c]: It’s here.

Comment (noroadsleft) [reply keyboards/reviung39/keymaps/toshi0383/keymap.c]: You need to delete the `layer_state_set_user` function on Lines 59-61. That function makes it so you can only access the Adjust layer if *both* the Lower and Raise layers are active. That function runs every time you change layers (either on or off). It's checking the layer state, finding that the Lower layer isn't active, and so it keeps you on the Raise layer.

If you delete the function, I think it should work.

File: keyboards/zinc/keymaps/toshi0383/keymap.c (hunk 9 above)
Comment (drashna) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: Di you still want to clean this up?

File: keyboards/reviung39/keymaps/toshi0383/keymap.c (hunk 10 above)
Comment (toshi0383) [reply keyboards/reviung39/keymaps/toshi0383/keymap.c]: Thank you so much!
Resolved in https://github.com/qmk/qmk_firmware/pull/7141/commits/53e49b9553ec08fc231cc137ff5f652a1ebaec67

File: keyboards/zinc/keymaps/toshi0383/keymap.c (hunk 9 above)
Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: Done in https://github.com/qmk/qmk_firmware/pull/7141/commits/8791b3d4a6db79bdd7779ea62984d59451ec665c

File: keyboards/zinc/keymaps/toshi0383/keymap.c [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+      } else {
+        layer_off(_ADJUST);
+      }
+      return false;
+      break;
+
+  }
+
+  return true;
+}
```
Comment (fauxpark) [inline keyboards/zinc/keymaps/toshi0383/keymap.c]: This whole function could be removed and the `custom_keycodes` enum replaced with simple `MO()` aliases, as you've done for your reviung39 keymap.

Comment (toshi0383) [reply keyboards/zinc/keymaps/toshi0383/keymap.c]: Thank you!
Done in https://github.com/qmk/qmk_firmware/pull/7141/commits/c902ae2287bd758111c2b49499c12b3906cc4ff1

**Comment by drashna** [comment]
Travis CI Error is unrelated.

**Comment by toshi0383** [comment]
🎉
Thanks for the reivew!

**Pull request merged**